spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/DefaultPubSubAcknowledger.java

```diff
@@ -27,23 +27,35 @@
  * ack IDs and subscription name.
  *
  * @author João André Martins
+ * @author Doug Hoard
  */
 class DefaultPubSubAcknowledger implements PubSubAcknowledger {
 
 	private SubscriberStub subscriberStub;
 
+	private boolean asynchronousAcknowledgement;
```

Comment:
I thought about an overloaded `(n)ackAsync()` and let end-user to choose.
The `boolean asyncAck`  could be as an option on the `PubSubSubscriberTemplate`. Or we could think about an appropriate `AckMode`...

Reply:
I am not opposed to this approach.